src/clj/harja/palvelin/palvelut/toteumat.clj

```diff
@@ -39,6 +39,22 @@
             (assoc :maara
                    (or (some-> % :maara double) 0)))))
 
+(defn vaadi-ei-jarjestelman-luoma [db toteuma-id]
+  (when toteuma-id
+    (let [jarjestelman-lisaama? (:jarjestelmanlisaama (first
+                                                        (toteumat-q/toteuma-jarjestelman-lisaama
+                                                          db {:toteuma toteuma-id})))]
+      (when jarjestelman-lisaama?
+        (throw (SecurityException. "Järjestelmän luomaa toteumaa ei voi muokata!"))))))
+
+(defn vaadi-toteuma-kuuluu-urakkaan [db toteuma-id vaitetty-urakka-id]
```

Comment:
Joissain SQL-kyselyissä on ehto, että toteuman urakan täytyy olla sama kuin annettu. Tämä on hyvä asia, mutta ei mielestäni ole riittävä, sillä sama asia pitäisi tarkistaa myös jokaisen toteumaan liittyvän asian kohdalla (toteuman tehtävät, reitti...). Nyt kun toteuma EI oikeasti kuulu väitettyyn urakkaan, sen tietoja ei päivitetä, mutta tehtävät ja reitti kyllä päivitetään. Näin ollen on mielestäni helpointa toteuman tallennuksen alussa tehdä kertaluontoinen tarkastus. Jos toteuma kuuluu väitettyyn urakkaan, niin sen muutkin tiedot voidaan päivittää turvallisesti.